src/main/java/gregtech/common/terminal/app/prospector/ProspectingTexture.java

```diff
@@ -44,17 +44,39 @@ public ProspectingTexture(int mode, int radius, boolean darkMode) {
     }
 
     public void updateTexture(PacketProspecting packet) {
+
         int playerChunkX = packet.posX >> 4;
         int playerChunkZ = packet.posZ >> 4;
-        playerI = packet.posX - (playerChunkX - this.radius + 1) * 16 - 1;
-        playerJ = packet.posZ - (playerChunkZ - this.radius + 1) * 16 - 1;
+        playerXGui = packet.posX - (playerChunkX - this.radius + 1) * 16;
+        playerYGui = packet.posZ - (playerChunkZ - this.radius + 1) * 16;
+
+        int ox;
+        if ((packet.chunkX >= 0 && playerChunkX >= 0) || (packet.chunkX <= 0 && playerChunkX <= 0)) {
+            ox = Math.abs(Math.abs(packet.chunkX) - Math.abs(playerChunkX));
+        } else {
+            ox = Math.abs(playerChunkX) + Math.abs(packet.chunkX);
+        }
+        if (playerChunkX > packet.chunkX) {
+            ox = -ox;
+        }
+
+        int oy;
+        if ((packet.chunkZ >= 0 && playerChunkZ >= 0) || (packet.chunkZ <= 0 && playerChunkZ <= 0)) {
+            oy = Math.abs(Math.abs(packet.chunkZ) - Math.abs(playerChunkZ));
+        } else {
+            oy = Math.abs(playerChunkZ) + Math.abs(packet.chunkZ);
+        }
+        if (playerChunkZ > packet.chunkZ) {
+            oy = -oy;
+        }
+
         if (this.mode == 1) {
-            map[packet.chunkX - (playerChunkX - radius + 1)][packet.chunkZ - (playerChunkZ - radius + 1)] = packet.map[0][0] == null ?
+            map[(this.radius - 1) + ox][(this.radius - 1) + oy] = packet.map[0][0] == null ?
```

Comment:
opening prospector occasionally crashes the game because of this line
full crash report:
https://pastebin.com/Wk9C7WkS